src/node_os.cc

```diff
@@ -214,14 +215,14 @@ static void GetInterfaceAddresses(const FunctionCallbackInfo<Value>& args) {
     if (interfaces[i].address.address4.sin_family == AF_INET) {
       uv_ip4_name(&interfaces[i].address.address4, ip, sizeof(ip));
       uv_ip4_name(&interfaces[i].netmask.netmask4, netmask, sizeof(netmask));
-      family = env->ipv4_string();
+      family = Integer::New(env->isolate(), 4);
     } else if (interfaces[i].address.address4.sin_family == AF_INET6) {
       uv_ip6_name(&interfaces[i].address.address6, ip, sizeof(ip));
       uv_ip6_name(&interfaces[i].netmask.netmask6, netmask, sizeof(netmask));
-      family = env->ipv6_string();
+      family = Integer::New(env->isolate(), 6);
```

Comment:
I'm concerned about these two specific changes, moving from IPv4  to 4 and IPv6 to 6. I'm 100% sure those will break some of my modules (Fastify v4).

Reply:
The reason for this change is that currently passing an address object to the DNS resolver, the `'IPv6'` string is converted to `0`.
Would it be reasonable to ask you to update the code that using that to expect one form or the other? Something like `info.family === 6 || info.family === 'IPv6'`? If not, we could make the DNS resolver accept a string and interpret `'IPv6'` same as `6`.

Reply:
I assume this change will hurt the community a lot

Reply:
Agreed here. I realize that Node v18 was a major version change, but changing the output of a widely used API (socket.address()) just so another core-Node API could match up with it instead of making the DNS resolver more flexible seems pretty overkill!

Reply:
Now that it's out in the wild, can confirm it caused some widespread breaking for us that's going to make node 18 a pain and very uncertain to roll out.

Reply:
> I assume this change will hurt the community a lot.

Can confirm completely unexpected breakage

Reply:
@aduh95 @mcollina Any chance to think about that change again and consider alternatives?

Reply:
very interesting, the proposal of this pr was to ensure the argument of the `dns` module, but these APIs have been deprecated in v17.0.0 https://nodejs.org/docs/latest-v17.x/api/deprecations.html#DEP0153

why not ensure the `dns` API arguments but change the `os` API, it did effects almost every node server module in the community.

it has been a month since 18.0.0 release and no one in the node team responded to this change. 

all the community can do is lock the node version and cannot update in a very long future

Reply:
maybe a new issue is more visible to the nodejs team and allows a better discussion ... so see https://github.com/nodejs/node/issues/43014 ... Please "thumbs up" it if you support to adjust this to be backward compatible. Thank you